### PR TITLE
DOCSUP-13431: Edit and translate to Russian (ZooKeeper replication and snowflake)

### PR DESCRIPTION
**Pull request by ka1bi4**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

Changelog category (leave one):
- Documentation for #27426 and #27704.

Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
- `snowflakeToDateTime`, `snowflakeToDateTime64`, `dateTimeToSnowflake`, `dateTime64ToSnowflake`
- ZooKeeper replication for users, roles, row policies, quotas and profiles.

### CODE REVIEW

File: docs/en/operations/server-configuration-parameters/settings.md [hunk 1]
```diff
@@ -1263,13 +1263,23 @@ The `user_directories` section can contain any number of items, the order of the
     <local_directory>
         <path>/var/lib/clickhouse/access/</path>
     </local_directory>
+</user_directories>
+```
+
+Users, roles, row policies, quotas and profiles can be also stored in ZooKeeper:
```
Comment (olgarev) [inline docs/en/operations/server-configuration-parameters/settings.md:1269]:
```suggestion
Users, roles, row policies, quotas, and profiles can be also stored in ZooKeeper:
```

File: docs/en/operations/server-configuration-parameters/settings.md [hunk 2]
```diff
@@ -1263,13 +1263,23 @@ The `user_directories` section can contain any number of items, the order of the
...
+        <path>/etc/clickhouse-server/users.xml</path>
+    </users_xml>
     <replicated>
         <zookeeper_path>/clickhouse/access/</zookeeper_path>
     </replicated>
 </user_directories>
 ```
 
-You can also specify settings `memory` — means storing information only in memory, without writing to disk, and `ldap` — means storing information on an LDAP server.
+You can also define sections memory `memory` — means storing information only in memory, without writing to disk, and `ldap` — means storing information on an LDAP server.
```
Comment (olgarev) [inline docs/en/operations/server-configuration-parameters/settings.md:1282]:
```suggestion
You can also define sections `memory` — means storing information only in memory, without writing to disk, and `ldap` — means storing information on an LDAP server.
```

File: docs/en/sql-reference/functions/type-conversion-functions.md [hunk 3]
```diff
@@ -1342,7 +1342,7 @@ Result:
 
 ## snowflakeToDateTime {#snowflakeToDateTime}
```
Comment (olgarev) [inline docs/en/sql-reference/functions/type-conversion-functions.md:1343]:
```suggestion
## snowflakeToDateTime {#snowflake-todatetime}
```

File: docs/en/sql-reference/functions/type-conversion-functions.md [hunk 4]
```diff
@@ -1378,7 +1378,7 @@ Result:
 
 ## snowflakeToDateTime64 {#snowflakeToDateTime64}
```
Comment (olgarev) [inline docs/en/sql-reference/functions/type-conversion-functions.md:1379]:
```suggestion
## snowflakeToDateTime64 {#snowflaketodatetime64}
```

File: docs/en/sql-reference/functions/type-conversion-functions.md [hunk 5]
```diff
@@ -1414,7 +1414,7 @@ Result:
 
 ## dateTimeToSnowflake {#dateTimeToSnowflake}
```
Comment (olgarev) [inline docs/en/sql-reference/functions/type-conversion-functions.md:1415]:
```suggestion
## dateTimeToSnowflake {#datetimetosnowflake}
```

File: docs/en/sql-reference/functions/type-conversion-functions.md [hunk 6]
```diff
@@ -1426,33 +1426,29 @@ dateTimeToSnowflake(value)
...
 
 ``` text
-
 ┌─dateTimeToSnowflake(dt)─┐
 │     1426860702823350272 │
 └─────────────────────────┘
 ```
 
-
 ## dateTime64ToSnowflake {#dateTime64ToSnowflake}
```
Comment (olgarev) [inline docs/en/sql-reference/functions/type-conversion-functions.md:1449]:
```suggestion
## dateTime64ToSnowflake {#datetime64tosnowflake}
```

File: docs/ru/operations/server-configuration-parameters/settings.md [hunk 7]
```diff
@@ -1218,7 +1219,20 @@ ClickHouse использует ZooKeeper для хранения метадан
...
+    <users_xml>
+        <path>/etc/clickhouse-server/users.xml</path>
+    </users_xml>
+    <replicated>
+        <zookeeper_path>/clickhouse/access/</zookeeper_path>
+    </replicated>
+</user_directories>
+```
+
+Также вы можете секцию `memory` — означает хранение информации только в памяти, без записи на диск, и `ldap` — означает хранения информации на [LDAP-сервере](https://en.wikipedia.org/wiki/Lightweight_Directory_Access_Protocol).
```
Comment (olgarev) [inline docs/ru/operations/server-configuration-parameters/settings.md:1235]:
```suggestion
Также вы можете добавить секции `memory` — означает хранение информации только в памяти, без записи на диск, и `ldap` — означает хранения информации на [LDAP-сервере](https://en.wikipedia.org/wiki/Lightweight_Directory_Access_Protocol).
```

File: docs/ru/sql-reference/functions/type-conversion-functions.md [hunk 8]
```diff
@@ -1338,3 +1338,144 @@ FROM numbers(3);
 │ 2,"good"                                  │
 └───────────────────────────────────────────┘
 ```
+
+## snowflakeToDateTime {#snowflakeToDateTime}
```
Comment (olgarev) [inline docs/ru/sql-reference/functions/type-conversion-functions.md:1342]:
```suggestion
## snowflakeToDateTime {#snowflaketodatetime}
```

File: docs/ru/sql-reference/functions/type-conversion-functions.md [hunk 9]
```diff
@@ -1338,3 +1338,144 @@ FROM numbers(3);
...
+Результат:
+
+``` text
+
+┌─snowflakeToDateTime(CAST('1426860702823350272', 'Int64'), 'UTC')─┐
+│                                              2021-08-15 10:57:56 │
+└──────────────────────────────────────────────────────────────────┘
+```
+
+## snowflakeToDateTime64 {#snowflakeToDateTime64}
```
Comment (olgarev) [inline docs/ru/sql-reference/functions/type-conversion-functions.md:1378]:
```suggestion
## snowflakeToDateTime64 {#snowflaketodatetime64}
```

File: docs/ru/sql-reference/functions/type-conversion-functions.md [hunk 10]
```diff
@@ -1338,3 +1338,144 @@ FROM numbers(3);
...
+Результат:
+
+``` text
+
+┌─snowflakeToDateTime64(CAST('1426860802823350272', 'Int64'), 'UTC')─┐
+│                                            2021-08-15 10:58:19.841 │
+└────────────────────────────────────────────────────────────────────┘
+```
+
+## dateTimeToSnowflake {#dateTimeToSnowflake}
```
Comment (olgarev) [inline docs/ru/sql-reference/functions/type-conversion-functions.md:1414]:
```suggestion
## dateTimeToSnowflake {#datetimetosnowflake}
```

File: docs/ru/sql-reference/functions/type-conversion-functions.md [hunk 11]
```diff
@@ -1338,3 +1338,144 @@ FROM numbers(3);
...
+
+Результат:
+
+``` text
+┌─dateTimeToSnowflake(dt)─┐
+│     1426860702823350272 │
+└─────────────────────────┘
+```
+
+## dateTime64ToSnowflake {#dateTime64ToSnowflake}
```
Comment (olgarev) [inline docs/ru/sql-reference/functions/type-conversion-functions.md:1448]:
```suggestion
## dateTime64ToSnowflake {#datetime64tosnowflake}
```

File: docs/en/sql-reference/functions/type-conversion-functions.md [hunk 12]
```diff
@@ -1342,7 +1342,7 @@ Result:
 
 ## snowflakeToDateTime {#snowflakeToDateTime}
 
-Extract time from snowflake id as DateTime format.
+Extracts time from Snowflake ID as [DateTime](../data-types/datetime.md) format.
```
Comment (olgarev) [inline docs/en/sql-reference/functions/type-conversion-functions.md:1345]:
```suggestion
Extracts time from [Snowflake ID](https://en.wikipedia.org/wiki/Snowflake_ID) as [DateTime](../data-types/datetime.md) format.
```

File: docs/en/sql-reference/functions/type-conversion-functions.md [hunk 13]
```diff
@@ -1378,7 +1378,7 @@ Result:
 
 ## snowflakeToDateTime64 {#snowflakeToDateTime64}
 
-Extract time from snowflake id as DateTime64 format.
+Extracts time from Snowflake ID as [DateTime64](../data-types/datetime64.md) format.
```
Comment (olgarev) [inline docs/en/sql-reference/functions/type-conversion-functions.md:1381]:
```suggestion
Extracts time from [Snowflake ID](https://en.wikipedia.org/wiki/Snowflake_ID) as [DateTime64](../data-types/datetime64.md) format.
```

File: docs/en/sql-reference/functions/type-conversion-functions.md [hunk 14]
```diff
@@ -1414,7 +1414,7 @@ Result:
 
 ## dateTimeToSnowflake {#dateTimeToSnowflake}
 
-Convert DateTime to the first snowflake id at the giving time.
+Converts [DateTime](../data-types/datetime.md) value to the first Snowflake ID at the giving time.
```
Comment (olgarev) [inline docs/en/sql-reference/functions/type-conversion-functions.md:1417]:
```suggestion
Converts [DateTime](../data-types/datetime.md) value to the first [Snowflake ID](https://en.wikipedia.org/wiki/Snowflake_ID) at the giving time.
```

File: docs/en/sql-reference/functions/type-conversion-functions.md [hunk 15]
```diff
@@ -1426,33 +1426,29 @@ dateTimeToSnowflake(value)
...
 ┌─dateTimeToSnowflake(dt)─┐
 │     1426860702823350272 │
 └─────────────────────────┘
 ```
 
-
 ## dateTime64ToSnowflake {#dateTime64ToSnowflake}
 
-Convert DateTime64 to the first snowflake id at the giving time.
+Convert [DateTime64](../data-types/datetime64.md) to the first Snowflake ID at the giving time.
```
Comment (olgarev) [inline docs/en/sql-reference/functions/type-conversion-functions.md:1451]:
```suggestion
Convert [DateTime64](../data-types/datetime64.md) to the first [Snowflake ID](https://en.wikipedia.org/wiki/Snowflake_ID) at the giving time.
```

File: docs/ru/sql-reference/functions/type-conversion-functions.md [hunk 16]
```diff
@@ -1338,3 +1338,144 @@ FROM numbers(3);
 │ 2,"good"                                  │
 └───────────────────────────────────────────┘
 ```
+
+## snowflakeToDateTime {#snowflakeToDateTime}
+
+Извлекает время из Snowflake ID в формате [DateTime](../data-types/datetime.md).
```
Comment (olgarev) [inline docs/ru/sql-reference/functions/type-conversion-functions.md:1344]:
```suggestion
Извлекает время из [Snowflake ID](https://en.wikipedia.org/wiki/Snowflake_ID) в формате [DateTime](../data-types/datetime.md).
```

File: docs/ru/sql-reference/functions/type-conversion-functions.md [hunk 17]
```diff
@@ -1338,3 +1338,144 @@ FROM numbers(3);
...
+``` text
+
+┌─snowflakeToDateTime(CAST('1426860702823350272', 'Int64'), 'UTC')─┐
+│                                              2021-08-15 10:57:56 │
+└──────────────────────────────────────────────────────────────────┘
+```
+
+## snowflakeToDateTime64 {#snowflakeToDateTime64}
+
+Извлекает время из Snowflake ID в формате [DateTime64](../data-types/datetime64.md).
```
Comment (olgarev) [inline docs/ru/sql-reference/functions/type-conversion-functions.md:1380]:
```suggestion
Извлекает время из [Snowflake ID](https://en.wikipedia.org/wiki/Snowflake_ID) в формате [DateTime64](../data-types/datetime64.md).
```

File: docs/ru/sql-reference/functions/type-conversion-functions.md [hunk 18]
```diff
@@ -1338,3 +1338,144 @@ FROM numbers(3);
...
+``` text
+
+┌─snowflakeToDateTime64(CAST('1426860802823350272', 'Int64'), 'UTC')─┐
+│                                            2021-08-15 10:58:19.841 │
+└────────────────────────────────────────────────────────────────────┘
+```
+
+## dateTimeToSnowflake {#dateTimeToSnowflake}
+
+Преобразует значение [DateTime](../data-types/datetime.md) в первый идентификатор Snowflake ID на текущий момент.
```
Comment (olgarev) [inline docs/ru/sql-reference/functions/type-conversion-functions.md:1416]:
```suggestion
Преобразует значение [DateTime](../data-types/datetime.md) в первый идентификатор [Snowflake ID](https://en.wikipedia.org/wiki/Snowflake_ID) на текущий момент.
```

File: docs/ru/sql-reference/functions/type-conversion-functions.md [hunk 19]
```diff
@@ -1338,3 +1338,144 @@ FROM numbers(3);
...
+
+``` text
+┌─dateTimeToSnowflake(dt)─┐
+│     1426860702823350272 │
+└─────────────────────────┘
+```
+
+## dateTime64ToSnowflake {#dateTime64ToSnowflake}
+
+Преобразует значение [DateTime64](../data-types/datetime64.md) в первый идентификатор Snowflake ID на текущий момент.
```
Comment (olgarev) [inline docs/ru/sql-reference/functions/type-conversion-functions.md:1450]:
```suggestion
Преобразует значение [DateTime64](../data-types/datetime64.md) в первый идентификатор [Snowflake ID](https://en.wikipedia.org/wiki/Snowflake_ID) на текущий момент.
```

File: docs/ru/operations/server-configuration-parameters/settings.md [hunk 20]
```diff
@@ -1200,12 +1200,13 @@ ClickHouse использует ZooKeeper для хранения метадан
 Секция конфигурационного файла,которая содержит настройки:
 -   Путь к конфигурационному файлу с предустановленными пользователями.
 -   Путь к файлу, в котором содержатся пользователи, созданные при помощи SQL команд.
+-   Путь к узлу ZooKeeper, где хранятся и реплицируются пользователи, созданные с помощью команд SQL (экспериментально).
```
Comment (alesapin) [inline docs/ru/operations/server-configuration-parameters/settings.md:1203]: Экспериментальная функциональность?

File: docs/en/sql-reference/functions/type-conversion-functions.md [hunk 21]
```diff
@@ -1426,33 +1426,29 @@ dateTimeToSnowflake(value)
 
 -   `value` — Date and time. [DateTime](../../sql-reference/data-types/datetime.md).
 
-
 **Returned value**
 
--   `value` converted to the `Int64` data type as the first snowflake id at that time.
+-   Input value converted to the [Int64](../data-types/int-uint.md) data type as the first Snowflake ID at that time.
```
Comment (alesapin) [inline docs/en/sql-reference/functions/type-conversion-functions.md:1431]: `first Snowflake ID` sounds unclear, but I don't know how to make it better.

**Review by alesapin — APPROVED** [review]

File: docs/ru/operations/server-configuration-parameters/settings.md (hunk 20 above)
Comment (ka1bi4) [reply docs/ru/operations/server-configuration-parameters/settings.md]: Thanks, fixed it.

File: docs/en/sql-reference/functions/type-conversion-functions.md (hunk 21 above)
Comment (ka1bi4) [reply docs/en/sql-reference/functions/type-conversion-functions.md:1431]: @alesapin 
May be, `the least Snowflake ID at that time`?

**Review by alesapin — APPROVED** [review]

**Pull request merged**